index.js

```diff
@@ -283,28 +286,84 @@ export const Config = {
 }
 
 export async function timer({ key }) {
+  const { state } = program;
   const [ owner, repo ] = key.split('/')
-  const result = await  client.activity.getEventsForRepo({ owner, repo });
-    for (let event of result.data) {
-      const { type, payload} = event;
-      if (type === 'IssuesEvent' && payload.action === 'opened') {
-        
-        // dispatch Event
-        const repoRef = root.users.one({ name: owner }).repos.one({ name: repo })
-        await repoRef.issueOpened.dispatch({
-          issue: repoRef.issues.one({ number: payload.issue.number })
-        });
-      };
-      if (type === 'PullRequestEvent' && payload.action === 'opened') {
-
-        // dispatch Event
-        const repoRef = root.users.one({ name: owner }).repos.one({ name: repo })
-        await repoRef.pullRequestOpened.dispatch({
-            issue: repoRef.issues.one({ number: payload.pull_request.number }),
-            pullRequest: repoRef.pullRequests.one({ number: payload.pull_request.number })
-        });
+  const result = await client.activity.getEventsForRepo({ owner, repo });
+
+  const data = result.data.reverse();
+  
+  const index = data
+    .findIndex(item => formatTime(item.created_at) > state.repos[key].lastEventTime);
+
+  const newEvents = data.splice(index);
```

Comment:
`splice` changes `data` which is a wasted effort. Use `slice(0, index)` and `>=` above

Reply:
Also, is better to reverse the filtered collection which will be, in many cases, many times smaller or even empty